include/dlaf/eigensolver/gen_to_std/impl.h

```diff
@@ -342,6 +403,260 @@ void GenToStd<backend, device, T>::call_L(comm::CommunicatorGrid grid, Matrix<T,
+    }
+  }
+}
+
+template <Backend backend, Device device, class T>
+void GenToStd<backend, device, T>::call_U(comm::CommunicatorGrid grid, Matrix<T, device>& mat_a,
+                                          Matrix<T, device>& mat_u) {
+  using namespace gentostd_u;
+  auto executor_hp = dlaf::getHpExecutor<backend>();
+  auto executor_np = dlaf::getNpExecutor<backend>();
+  auto executor_mpi = dlaf::getMPIExecutor<backend>();
+
+  // Set up MPI executor pipelines
+  common::Pipeline<comm::Communicator> mpi_row_task_chain(grid.rowCommunicator());
+  common::Pipeline<comm::Communicator> mpi_col_task_chain(grid.colCommunicator());
+
+  const comm::Index2D this_rank = grid.rank();
+
+  const matrix::Distribution& distr = mat_a.distribution();
+  // Number of tile (rows = cols)
+  SizeType nrtile = mat_a.nrTiles().rows();
+
+  constexpr std::size_t n_workspaces = 2;
+  common::RoundRobin<matrix::Panel<Coord::Row, T, device>> a_panels(n_workspaces, distr);
+  common::RoundRobin<matrix::Panel<Coord::Col, T, device>> a_panelsT(n_workspaces, distr);
+  common::RoundRobin<matrix::Panel<Coord::Row, T, device>> u_panels(n_workspaces, distr);
+  common::RoundRobin<matrix::Panel<Coord::Col, T, device>> u_panelsT(n_workspaces, distr);
+
+  for (SizeType k = 0; k < nrtile; ++k) {
+    const GlobalTileIndex kk{k, k};
+    const comm::Index2D kk_rank = distr.rankGlobalTile(kk);
+    const auto kt = k + 1;
+
+    const LocalTileSize kk_offset{
+        distr.nextLocalTileFromGlobalTile<Coord::Row>(k),
+        distr.nextLocalTileFromGlobalTile<Coord::Col>(k),
+    };
+
+    const LocalTileSize at_offset{
+        distr.nextLocalTileFromGlobalTile<Coord::Row>(kt),
+        distr.nextLocalTileFromGlobalTile<Coord::Col>(kt),
+    };
+
+    auto& u_panel = u_panels.nextResource();
+    auto& u_panelT = u_panelsT.nextResource();
+    auto& a_panel = a_panels.nextResource();
+    auto& a_panelT = a_panelsT.nextResource();
+    u_panel.setRangeStart({k, k});
+    u_panelT.setRangeStart({k, k});
+    a_panelT.setRange({0, 0}, {k, k});
+
+    if (k == nrtile - 1) {
+      u_panel.setHeight(distr.tileSize(kk).rows());
+      a_panelT.setWidth(distr.tileSize(kk).cols());
+    }
+
+    if (kk_rank.row() == this_rank.row()) {
+      for (SizeType i_local = kk_offset.cols(); i_local < distr.localNrTiles().cols(); ++i_local) {
+        const LocalTileIndex ki_panel(Coord::Col, i_local);
+        const LocalTileIndex ki(kk_offset.rows(), i_local);
+
+        u_panel.setTile(ki_panel, mat_u.read(ki));
+      }
+    }
+
+    broadcast(executor_mpi, kk_rank.row(), u_panel, u_panelT, mpi_row_task_chain, mpi_col_task_chain);
+
+    // continue update previous panels
+    // Note: The tasks of the final huge TRSM of the HEGST step have been reshuffled to avoid extra
+    //       communication of the matrix U.
+    //       During k-th iteration only the tasks involving the k-th panel of U are executed.
+    //       Therefore, all previous panel have to be updated at each step.
```

Comment:
```suggestion
    //       Therefore, all previous panels have to be updated at each step.
```
Please fix it in the lower case as well.